domain/src/main/java/uk/gov/hmcts/ccd/definition/store/domain/validation/displaycontextparameter/AbstractDisplayContextParameterValidator.java

```diff
@@ -0,0 +1,136 @@
+package uk.gov.hmcts.ccd.definition.store.domain.validation.displaycontextparameter;
+
+import com.google.common.base.Strings;
+import uk.gov.hmcts.ccd.definition.store.domain.displaycontextparameter.DisplayContextParameter;
+import uk.gov.hmcts.ccd.definition.store.domain.displaycontextparameter.DisplayContextParameterType;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.SimpleValidationError;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationError;
+import uk.gov.hmcts.ccd.definition.store.domain.validation.ValidationResult;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Optional;
+
+public abstract class AbstractDisplayContextParameterValidator<T extends Serializable> {
```

Comment:
We could consider abbreviating DisplayContextParameter with DCP to shorten the class and subclass names.